### ISSUE TITLE
Add debug options to the GUI

### ISSUE THREAD

**Issue by leberwurstsaft** [textext/textext#35]
I think we should add some kind of debug option toggles right into the normal release. Or alternatively some way to read an environment variable that governs the debug flag(s). The manual copying and pasting of debug versions is 1. tedious, 2. prone to errors and a source of confusion if it isn’t correctly rolled back/replaced afterwards.

**Comment by sizmailov** [comment]
I tried to rework error handling. It seems like the right place to discuss/criticize the approach. 

Changes:
 - Do top-level error collection
 - Do logging 
    - One channel directed to a log file (not currently work on Windows, due to strange permissions problem, tried several places with no luck. Any suggestions? ) 
    - Another one is to be shown to end user
 - Hierarchy of exceptions is introduced 
 - Use context managers instead of `try: ... finalize: ...` to 
    - done for temp directories
    - done for change directories 

For details see https://github.com/sizmailov/textext/tree/do_propper_error_handling 
Current version solves #17

**Comment by jcwinkler** [comment]
This is awesome!

I tested it under my Windows installation and it ran out of the box, the log file was created successfully. I provoked some errors and they were properly logged. 

Under Windows the extension files usually goe to `c:\Users\USERNAME\AppData\Roaming\inkscape\extensions\textext\` and so the log file does. The user has write permissions there, otherwise he would not have been able to install the extension. Question is why it did not work on your System, Sergei.

One comment to the section

https://github.com/sizmailov/textext/blob/8e25ede3c433f4ad8db28becbd6fb138bbd63a53/extension/textext/__init__.py#L647-L675

If the command is not found this is treated as non fatal. This should be handled in more detail because such a scenario is fatal.

**Comment by sizmailov** [comment]
Sorry, I copied to TexText to system extensions folder instead of appdata, so just my mistake. 

I'm not 100% happy with current state of the code. Let me clarify it a bit:

Fatal errors are errors which disables normal functioning of extension, for example if latex is missing. 
Non fatal are errors which breaks normal execution, but we able to recover. For example: imagemagick is missing. 

Both commands use `exec_command` function. Inside of exec_command we are unable to tell is this command crucial for TexText or optional. I took minimum of errors. 

In fact we should throw other exceptions inside of `exec_command` which does not state importance of the error, but only the fact of error, like "CommandNotFoundError" and "CommandExecutionError". The caller would be responsible to turn them to Fatal/NonFatal. 

Right now I turn some of NonFatal errors into FatalErrors, which is better in my tests, than other way around.
